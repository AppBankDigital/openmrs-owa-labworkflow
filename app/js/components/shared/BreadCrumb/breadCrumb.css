/*
 * This Source Code Form is subject to the terms of the Mozilla Public License,
 * v. 2.0. If a copy of the MPL was not distributed with this file, You can
 * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
 * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
 *
 * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
 * graphic logo is a trademark of OpenMRS Inc.
 */

 .breadcrumb-parent {
    background: none;
    margin: 0px;
    padding: 0px;
    color: #363463;
    text-transform: capitalize;
}

.breadcrumb-item {
    color: #363463;
    margin-left: 3px;
    font-size: 12px;
}

.breadcrumb-parent>.separator {
    font-size: 15px;
}

.breadcrumb-parent .glyphicon {
    color: #555;
}
