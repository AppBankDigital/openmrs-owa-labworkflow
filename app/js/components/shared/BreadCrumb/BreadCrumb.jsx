/*
 * This Source Code Form is subject to the terms of the Mozilla Public License,
 * v. 2.0. If a copy of the MPL was not distributed with this file, You can
 * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
 * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
 *
 * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
 * graphic logo is a trademark of OpenMRS Inc.
 */

import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link, withRouter } from 'react-router-dom';

import './breadCrumb.css';

class BreadCrumb extends Component {
  constructor(props) {
    super(props);

    this.state = {
      currentTab: '',
    };
  }

  componentWillMount() {
    const { history } = this.props;
    history.listen((location) => {
      const currentTab = location.pathname.split('/').pop();
      this.setState({
        currentTab,
      });
    });
  }


  render() {
    const {
      currentTab,
    } = this.state;
    let familyName = '';
    const { patientHeaderDetail, history } = this.props;
    if (typeof patientHeaderDetail !== 'undefined' && patientHeaderDetail.hasOwnProperty('person')) {
      familyName = patientHeaderDetail.person.personName.familyName;
    }
    let outputBreadcrumb;

    const homePageBreadcrumb = (
      <span>
        <Link to="/">
          <span
            className="glyphicon glyphicon-chevron-right breadcrumb-item separator"
            aria-hidden="true" />
          <span className="title breadcrumb-item">
            <u>
            Lab
            </u>
          </span>
        </Link>
        <Link to="/">
          <span
            className="glyphicon glyphicon-chevron-right breadcrumb-item separator"
            aria-hidden="true" />
          <span className="title breadcrumb-item">
            <u>Lab Test Results</u>
          </span>
        </Link>
      </span>
    );

    switch (currentTab) {
      case 'LabResultEntry':
        outputBreadcrumb = (
          <span>
            <span>{ homePageBreadcrumb }</span>
            <Link to="LabResultEntry">
              <span
                className="glyphicon glyphicon-chevron-right breadcrumb-item separator"
                aria-hidden="true" />
              <span className="title breadcrumb-item">
                {familyName || ''}
                {' '}
Lab Result
              </span>
            </Link>
          </span>
        );
        break;

      case 'labresults':
        outputBreadcrumb = (
          <span>
            <Link to="LabResultEntry">
              <span
                className="glyphicon glyphicon-chevron-right breadcrumb-item separator"
                aria-hidden="true" />
              <span className="title breadcrumb-item">
                <u>{`${familyName} test`}</u>
              </span>
            </Link>
            <Link to="labresults">
              <span
                className="glyphicon glyphicon-chevron-right breadcrumb-item separator"
                aria-hidden="true" />
              <span className="title breadcrumb-item">
                Patient Labs
              </span>
            </Link>
          </span>
        );
        break;

      case 'labtrends':
        outputBreadcrumb = (
          <span>
            <Link to="LabResultEntry">
              <span
                className="glyphicon glyphicon-chevron-right breadcrumb-item separator"
                aria-hidden="true" />
              <span className="title breadcrumb-item">
                <u>{`${familyName} test`}</u>
              </span>
            </Link>
            <Link to="labresults">
              <span
                className="glyphicon glyphicon-chevron-right breadcrumb-item separator"
                aria-hidden="true" />
              <span className="title breadcrumb-item">
                <u>Patient Labs</u>
              </span>
            </Link>
            <Link to="labresults">
              <span
                className="glyphicon glyphicon-chevron-right breadcrumb-item separator"
                aria-hidden="true" />
              <span className="title breadcrumb-item">
                {`${history.location.state.display} Trend`}
              </span>
            </Link>
          </span>
        );
        break;

      default:
        outputBreadcrumb = homePageBreadcrumb;
    }

    return (
      <div className="breadcrumb">
        <a href="../../" className="breadcrumb-item">
          <span className="glyphicon glyphicon-home breadcrumb-item" aria-hidden="true" />
        </a>
        { outputBreadcrumb }
      </div>
    );
  }
}

BreadCrumb.propTypes = {
  history: PropTypes.shape({}).isRequired,
  patientHeaderDetail: PropTypes.shape({}),
};

const mapStateToProps = ({
  patients,
  selectedPatient,
}) => ({
  patientHeaderDetail: patients[selectedPatient],
});


export default withRouter(connect(mapStateToProps)(BreadCrumb));
