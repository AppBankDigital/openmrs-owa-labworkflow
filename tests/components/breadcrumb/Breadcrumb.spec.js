import React from 'react';
import { shallowToJson } from 'enzyme-to-json';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';

import BreadCrumb from '../../../app/js/components/shared/BreadCrumb/BreadCrumb';

let mountedComponent;
const getComponent = () => {
  if (!mountedComponent) {
    mountedComponent = mount(
      <MemoryRouter initialEntries={[ { pathname: '/', key: 'testKey' } ]}>
        <BreadCrumb />
      </MemoryRouter>,
    );
  }
  return mountedComponent;
};

describe('Breadcrumb component', () => {
  it('should take a snapshot on initial render', () => {
    expect(getComponent()).toMatchSnapshot();
  });
});
