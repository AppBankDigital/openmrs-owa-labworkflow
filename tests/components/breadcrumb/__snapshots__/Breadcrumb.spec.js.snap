// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb component should take a snapshot on initial render 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
        "pathname": "/",
      },
    ]
  }
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "textComponent": "span",
    }
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(BreadCrumb)>
      <Route>
        <BreadCrumb
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "canGo": [Function],
              "createHref": [Function],
              "entries": Array [
                Object {
                  "hash": "",
                  "key": "testKey",
                  "pathname": "/",
                  "search": "",
                },
              ],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "index": 0,
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "key": "testKey",
                "pathname": "/",
                "search": "",
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          location={
            Object {
              "hash": "",
              "key": "testKey",
              "pathname": "/",
              "search": "",
            }
          }
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
        >
          <div
            className="breadcrumb"
          >
            <a
              className="breadcrumb-item"
              href="../../"
            >
              <span
                aria-hidden="true"
                className="glyphicon glyphicon-home breadcrumb-item"
              />
            </a>
            <Link
              replace={false}
              to="/"
            >
              <a
                href="/"
                onClick={[Function]}
              >
                <span
                  aria-hidden="true"
                  className="glyphicon glyphicon-chevron-right breadcrumb-item separator"
                />
                <span
                  className="title breadcrumb-item"
                >
                  <u>
                    <FormattedMessage
                      defaultMessage="Lab workflow"
                      description="the homepage breadcrumb title"
                      id="app.breadCrumb.homePageBreadcrumb.title"
                      values={Object {}}
                    >
                      <span>
                        Lab workflow
                      </span>
                    </FormattedMessage>
                  </u>
                </span>
              </a>
            </Link>
          </div>
        </BreadCrumb>
      </Route>
    </withRouter(BreadCrumb)>
  </Router>
</MemoryRouter>
`;
